front/lib/api/poke/plugins/data_sources/operations.ts

```diff
@@ -0,0 +1,73 @@
+import { assertNever, ConnectorsAPI, Err, Ok } from "@dust-tt/types";
+
+import config from "@app/lib/api/config";
+import { createPlugin } from "@app/lib/api/poke/types";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import logger from "@app/logger/logger";
+
+const OPERATIONS = ["STOP", "PAUSE", "UNPAUSE", "RESUME", "SYNC"] as const;
+
+const doOperation = (op: (typeof OPERATIONS)[number], connectorId: string) => {
+  const connectorsAPI = new ConnectorsAPI(
+    config.getConnectorsAPIConfig(),
+    logger
+  );
+
+  switch (op) {
+    case "STOP":
+      return connectorsAPI.stopConnector(connectorId);
+    case "PAUSE":
+      return connectorsAPI.pauseConnector(connectorId);
+    case "UNPAUSE":
+      return connectorsAPI.unpauseConnector(connectorId);
+    case "RESUME":
+      return connectorsAPI.resumeConnector(connectorId);
+    case "SYNC":
+      return connectorsAPI.syncConnector(connectorId);
+    default:
+      assertNever(op);
+  }
+};
+
+export const connectorOperations = createPlugin(
+  {
+    id: "maintenance-operation",
+    name: "Maintenance operation",
+    description: "Execute a maintenance operation on connector",
+    resourceTypes: ["data_sources"],
+    args: {
+      op: {
+        type: "enum",
+        label: "Operation",
+        description: "Select operation to execute",
+        values: OPERATIONS,
+      },
+    },
+  },
+  async (auth, dataSourceId, args) => {
+    if (!dataSourceId) {
```

Comment:
This should never happen, right? I'm pretty sure the type enforce this!

Reply:
the type is `resourceId: string | undefined` , so yes, could happen